Source/monster.cpp

```diff
@@ -2207,7 +2184,7 @@ bool AiPlanPath(int i)
 	}
 
 	bool clear = LineClear(
-	    std::bind(MonsterIsTileAvailable, i, std::placeholders::_1),
+	    [monster](Point position) { return IsTileAvailable(monster, position); },

```

Comment:
This captures `monster` by-value (copies it). Use `&monster` instead to capture by reference.